migrations/redactProjectEventVariables.go

```diff
@@ -0,0 +1,182 @@
+package migrations
+
+import (
+	"context"
+	"os"
+	"strconv"
+
+	"github.com/evergreen-ci/evergreen/model"
+	"github.com/evergreen-ci/evergreen/model/event"
+	"github.com/mongodb/anser/bsonutil"
+	"github.com/mongodb/grip"
+	"github.com/pkg/errors"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+const (
+	redactProjectEventSecretsName = "redactProjectEventSecrets"
+
+	// Optional env vars to control what project ID to start at for branch
+	// project and repo refs.
+	startAtProjectIDEnvVar = "START_AT_PROJECT_ID"
+	startAtRepoIDEnvVar    = "START_AT_REPO_ID"
+	// Limit how many projects can be processed by the job.
+	projectLimitEnvVar = "PROJECT_LIMIT"
+)
+
+var (
+	eventIDKey   = bsonutil.MustHaveTag(event.EventLogEntry{}, "ID")
+	eventTypeKey = bsonutil.MustHaveTag(event.EventLogEntry{}, "EventType")
+)
+
+func init() {
+	Registry.registerMigration(redactProjectEventSecretsName, newRedactProjectEventSecrets)
+}
+
+// redactProjectEventSecrets is a migration to retroactively redact project
+// secret values from project modifications in the event log.
+type redactProjectEventSecrets struct {
+	database string
+}
+
+func newRedactProjectEventSecrets(opts MigrationOptions) (Migration, error) {
+	catcher := grip.NewBasicCatcher()
+	catcher.Add(errors.Wrap(opts.validate(), "invalid options"))
+
+	return &redactProjectEventSecrets{
+		database: opts.Database,
+	}, catcher.Resolve()
+}
+
+func (c *redactProjectEventSecrets) Execute(ctx context.Context, client *mongo.Client) error {
+	collInfos := []struct {
+		name      string
+		startAtID string
+	}{
+		{
+			name:      "project_ref",
+			startAtID: os.Getenv(startAtProjectIDEnvVar),
+		},
+		{
+			name:      "repo_ref",
+			startAtID: os.Getenv(startAtRepoIDEnvVar),
+		},
+	}
+
+	var projectLimit int
+	if projectLimitStr := os.Getenv(projectLimitEnvVar); projectLimitStr != "" {
+		var err error
+		projectLimit, err = strconv.Atoi(projectLimitStr)
+		if err != nil {
+			return errors.Wrapf(err, "parsing project limit '%s' from env var '%s'", projectLimitStr, projectLimitEnvVar)
+		}
+	}
+
+	numProjectsProcessed := 0
+	for _, collInfo := range collInfos {
+		q := bson.M{}
+		if collInfo.startAtID != "" {
+			q["_id"] = bson.M{"$gte": collInfo.startAtID}
+			grip.Infof("Starting at project '%s' in collection '%s'\n", collInfo.startAtID, collInfo.name)
+		}
+		// Sort by _id to iterate in a predictable order. This makes it easier to
+		// resume from a specific project if the migration fails partway through.
+		findOpts := options.Find().SetSort(bson.M{"_id": 1}).SetProjection(bson.M{"_id": 1})
+		cur, err := client.Database(c.database).Collection(collInfo.name).Find(ctx, q, findOpts)
+		if err != nil {
+			return errors.Wrap(err, "finding project refs")
```

Comment:
mega nit: format collection name here since it can also be repo refs

Reply:
Done.